{"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":293,"status":"ok","timestamp":1629633123969,"user":{"displayName":"田中太郎","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GhWIE75kCAmONK6kifUrIQFZKDp1j3Tf2jyQq4T=s64","userId":"14683603613507086053"},"user_tz":-540},"id":"hXgs5B4wVfJS","outputId":"b7b6ec04-f4a7-416b-9b7d-6ba70ed08ec9"},"outputs":[{"name":"stdout","output_type":"stream","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n","Sun Aug 22 11:52:03 2021       \n","+-----------------------------------------------------------------------------+\n","| NVIDIA-SMI 470.57.02    Driver Version: 460.32.03    CUDA Version: 11.2     |\n","|-------------------------------+----------------------+----------------------+\n","| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |\n","| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |\n","|                               |                      |               MIG M. |\n","|===============================+======================+======================|\n","|   0  Tesla P100-PCIE...  Off  | 00000000:00:04.0 Off |                    0 |\n","| N/A   38C    P0    28W / 250W |      0MiB / 16280MiB |      0%      Default |\n","|                               |                      |                  N/A |\n","+-------------------------------+----------------------+----------------------+\n","                                                                               \n","+-----------------------------------------------------------------------------+\n","| Processes:                                                                  |\n","|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |\n","|        ID   ID                                                   Usage      |\n","|=============================================================================|\n","|  No running processes found                                                 |\n","+-----------------------------------------------------------------------------+\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')\n","!nvidia-smi"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"background_save":true,"base_uri":"https://localhost:8080/"},"id":"vrDEHNewVljI"},"outputs":[{"name":"stdout","output_type":"stream","text":["/content/drive/My Drive/alpha/quoridor\n","/usr/local/lib/python3.7/dist-packages/keras/utils/generic_utils.py:497: CustomMaskWarning: Custom mask layers require a config and must override get_config. When loading, the custom mask layer must be passed to the custom_objects argument.\n","  category=CustomMaskWarning)\n","********** Train for 1 / 200 **********\n","Self Play: 1/1000 =\u003e 0\n","Self Play: 2/1000 =\u003e 0\n","Self Play: 3/1000 =\u003e 0\n","Self Play: 4/1000 =\u003e 0\n","Self Play: 5/1000 =\u003e 0\n","Self Play: 6/1000 =\u003e 0\n","Self Play: 7/1000 =\u003e 0\n","Self Play: 8/1000 =\u003e 0\n","Self Play: 9/1000 =\u003e 0\n","Self Play: 10/1000 =\u003e 0\n","Self Play: 11/1000 =\u003e 0\n","Self Play: 12/1000 =\u003e 0\n","Self Play: 13/1000 =\u003e 0\n","Self Play: 14/1000 =\u003e 0\n","Self Play: 15/1000 =\u003e 0\n","Self Play: 16/1000 =\u003e 0\n","Self Play: 17/1000 =\u003e 0\n","Self Play: 18/1000 =\u003e 0\n","Self Play: 19/1000 =\u003e 0\n","Self Play: 20/1000 =\u003e 0\n","Self Play: 21/1000 =\u003e 0\n","Self Play: 22/1000 =\u003e 0\n","Self Play: 23/1000 =\u003e 0\n","Self Play: 24/1000 =\u003e Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py\", line 1007, in func_graph_from_py_func\n","    func_outputs = python_func(*func_args, **func_kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/data/ops/dataset_ops.py\", line 4195, in wrapped_fn\n","    ret = wrapper_helper(*args)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/data/ops/dataset_ops.py\", line 4125, in wrapper_helper\n","    ret = autograph.tf_convert(self._func, ag_ctx)(*nested_args)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/autograph/impl/api.py\", line 692, in wrapper\n","    return converted_call(f, args, kwargs, options=options)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/autograph/impl/api.py\", line 382, in converted_call\n","    return _call_unconverted(f, args, kwargs, options)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/autograph/impl/api.py\", line 463, in _call_unconverted\n","    return f(*args, **kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/keras/engine/data_adapter.py\", line 277, in permutation\n","    indices = tf.range(num_samples, dtype=tf.int64)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/util/dispatch.py\", line 206, in wrapper\n","    return target(*args, **kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/math_ops.py\", line 2064, in range\n","    return gen_math_ops._range(start, limit, delta, name=name)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/gen_math_ops.py\", line 7370, in _range\n","    \"Range\", start=start, limit=limit, delta=delta, name=name)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/op_def_library.py\", line 377, in _apply_op_helper\n","    with g.as_default(), ops.name_scope(name) as scope:\n","  File \"/usr/lib/python3.7/contextlib.py\", line 112, in __enter__\n","    return next(self.gen)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py\", line 413, in inner_cm\n","    with outer_cm as g:\n","  File \"/usr/lib/python3.7/contextlib.py\", line 112, in __enter__\n","    return next(self.gen)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/ops.py\", line 5617, in get_controller\n","    context.context().context_switches.push(default.building_function,\n","KeyboardInterrupt\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"train.py\", line 30, in \u003cmodule\u003e\n","    i + 1,\n","  File \"/content/drive/My Drive/alpha/quoridor/game.py\", line 188, in selfmatch\n","    hist_input, hist_pv, value = self.match(net, algo, simulations, gamma)\n","  File \"/content/drive/My Drive/alpha/quoridor/game.py\", line 138, in match\n","    probs = algo.get_probs(net, board, simulations, gamma)\n","  File \"/content/drive/My Drive/alpha/quoridor/montecarlo.py\", line 57, in get_probs\n","    root.eval(net)\n","  File \"/content/drive/My Drive/alpha/quoridor/mcts_node.py\", line 187, in eval\n","    value = -self._next_child().eval(net)\n","  File \"/content/drive/My Drive/alpha/quoridor/mcts_node.py\", line 187, in eval\n","    value = -self._next_child().eval(net)\n","  File \"/content/drive/My Drive/alpha/quoridor/mcts_node.py\", line 187, in eval\n","    value = -self._next_child().eval(net)\n","  [Previous line repeated 120 more times]\n","  File \"/content/drive/My Drive/alpha/quoridor/mcts_node.py\", line 176, in eval\n","    policy, value = self.predict(net, self.board)\n","  File \"/content/drive/My Drive/alpha/quoridor/mcts_node.py\", line 141, in predict\n","    y = net.model.predict(x, batch_size=1)\n","  File \"/usr/local/lib/python3.7/dist-packages/keras/engine/training.py\", line 1730, in predict\n","    steps_per_execution=self._steps_per_execution)\n","  File \"/usr/local/lib/python3.7/dist-packages/keras/engine/data_adapter.py\", line 1383, in get_data_handler\n","    return DataHandler(*args, **kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/keras/engine/data_adapter.py\", line 1150, in __init__\n","    model=model)\n","  File \"/usr/local/lib/python3.7/dist-packages/keras/engine/data_adapter.py\", line 286, in __init__\n","    indices_dataset = indices_dataset.map(permutation).prefetch(1)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/data/ops/dataset_ops.py\", line 1861, in map\n","    return MapDataset(self, map_func, preserve_cardinality=True)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/data/ops/dataset_ops.py\", line 4985, in __init__\n","    use_legacy_function=use_legacy_function)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/data/ops/dataset_ops.py\", line 4218, in __init__\n","    self._function = fn_factory()\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\", line 3151, in get_concrete_function\n","    *args, **kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\", line 3116, in _get_concrete_function_garbage_collected\n","    graph_function, _ = self._maybe_define_function(args, kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\", line 3463, in _maybe_define_function\n","    graph_function = self._create_graph_function(args, kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/eager/function.py\", line 3308, in _create_graph_function\n","    capture_by_value=self._capture_by_value),\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/func_graph.py\", line 1048, in func_graph_from_py_func\n","    func_graph.variables = variables\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/auto_control_deps.py\", line 384, in __exit__\n","    if (op_def_registry.get(op.type) is None or\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/ops.py\", line 2453, in type\n","    return pywrap_tf_session.TF_OperationOpType(self._c_op)\n","KeyboardInterrupt\n"]}],"source":["%cd \"/content/drive/My Drive/alpha/quoridor\"\n","!python train.py"]}],"metadata":{"accelerator":"GPU","colab":{"authorship_tag":"ABX9TyOJVxYf80Vxh+3/ijCqzTHq","collapsed_sections":[],"machine_shape":"hm","name":"train.ipynb","version":""},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}